src/charm.py

```diff
@@ -215,13 +230,44 @@ def ready_to_start(self) -> bool:
         Returns:
             True if ZK is related and `sync` user has been added. False otherwise.
         """
+        # SSL must be enabled for Kafka and ZK or disabled for both
+        if self.tls.enabled ^ (
```

Comment:
Switch the ^ to 'and not'

Reply:
In this case I need to skip the if when both conditions are true, or both are false. That situation is a XOR gate, in python `^`.
Using `and not` would make:  `(kafka.tls = False) and ¬(zk.tls = True)` evaluate to false and skip the warning.